lib/datadog/tracing/contrib/rack/header_tagging.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Rack
+        # Matches Rack-style headers with a matcher and sets matching headers into a span.
+        module HeaderTagging
+          def self.tag_request_headers(span, env, configuration)
+            # Wrap env in a case-insensitive Rack-style accessor.
+            headers = env.is_a?(Header::RequestHeaderCollection) ? env : Header::RequestHeaderCollection.new(env)
+
+            # Use global DD_TRACE_HEADER_TAGS if integration-level configuration is not provided
+            tags = if configuration.using_default?(:headers) && Datadog.configuration.tracing.header_tags.configured?
```

Comment:
If you apply this comment https://github.com/DataDog/dd-trace-rb/pull/2935/files#r1270699898

You could change the code to: 
```suggestion
            tags = if configuration.using_default?(:headers) && !Datadog.configuration.tracing.using_default(:header_tags)
```

And we can get rid of the `configured?` method all together

Reply:
Let's try it!